tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9573,45 +9573,64 @@ struct int2 {
 };
 
 bool VerifyQuadAnyAllResults(int2 *Res) {
-    int Idx = 0;
-    for ( ; Idx < 4; ++Idx) {
-      if (Res[Idx].x != 2) return false;
-      if (Res[Idx].y != 4) return false;
-    }
-    for ( ; Idx < 60; ++Idx) {
-      if (Res[Idx].x != 1) return false;
-      if (Res[Idx].y != 4) return false;
-    }
-    for ( ; Idx < 64; ++Idx) {
-      if (Res[Idx].x != 1) return false;
-      if (Res[Idx].y != 3) return false;
-    }
-    return true;
+  int Idx = 0;
+  for (; Idx < 4; ++Idx) {
+    if (Res[Idx].x != 2)
+      return false;
+    if (Res[Idx].y != 4)
+      return false;
+  }
+  for (; Idx < 60; ++Idx) {
+    if (Res[Idx].x != 1)
+      return false;
+    if (Res[Idx].y != 4)
+      return false;
+  }
+  for (; Idx < 64; ++Idx) {
+    if (Res[Idx].x != 1)
+      return false;
+    if (Res[Idx].y != 3)
+      return false;
+  }
+  return true;
 }
 
 TEST_F(ExecutionTest, QuadAnyAll) {
-  WEX::TestExecution::SetVerifyOutput verifySettings(WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
   CComPtr<IStream> pStream;
   ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
 
-  std::shared_ptr<st::ShaderOpSet> ShaderOpSet = std::make_shared<st::ShaderOpSet>();
+  std::shared_ptr<st::ShaderOpSet> ShaderOpSet =
+      std::make_shared<st::ShaderOpSet>();
   st::ParseShaderOpSetFromStream(pStream, ShaderOpSet.get());
-  st::ShaderOp* pShaderOp = ShaderOpSet->GetShaderOp("QuadAnyAll");
+  st::ShaderOp *pShaderOp = ShaderOpSet->GetShaderOp("QuadAnyAll");
 
   LPCSTR args = "/Od";
 
   if (args[0]) {
-    for (st::ShaderOpShader& S : pShaderOp->Shaders)
+    for (st::ShaderOpShader &S : pShaderOp->Shaders)
       S.Arguments = args;
   }
 
   bool Skipped = true;
-  D3D_SHADER_MODEL TestShaderModels[] = { D3D_SHADER_MODEL_6_0, D3D_SHADER_MODEL_6_7 };
+  D3D_SHADER_MODEL TestShaderModels[] = {D3D_SHADER_MODEL_6_0,
+                                         D3D_SHADER_MODEL_6_5};
   for (unsigned i = 0; i < _countof(TestShaderModels); i++) {
     D3D_SHADER_MODEL sm = TestShaderModels[i];
-    LogCommentFmt(L"\r\nVerifying QuadAny/QuadAll using Wave intrinsics in shader model 6.%1u", ((UINT)sm & 0x0f));
-
-    if (sm == D3D_SHADER_MODEL_6_7) {
+    LogCommentFmt(L"\r\nVerifying QuadAny/QuadAll using Wave intrinsics in "
+                  L"shader model 6.%1u",
+                  ((UINT)sm & 0x0f));
+    if (sm == D3D_SHADER_MODEL_6_5) {
+      pShaderOp->PS = "PS";
+      pShaderOp->VS = "VS";
+      pShaderOp->MS = "MS";
+      pShaderOp->AS = "AS";
+    } else if (sm == D3D_SHADER_MODEL_6_7) {
+      pShaderOp->PS = "PS";
+      pShaderOp->VS = "VS";
+      pShaderOp->AS = "AS67";
+      pShaderOp->MS = "MS67";
       pShaderOp->CS = "CS67";
```

Comment:
In my experience, the checks for these were pointer compares, so I had to search for the matching names in the shaders in ShaderOp. Maybe that's changed?

Reply:
Yup, just ran into this during testing. See #4293 for the fix for the CS67 case. The same logic can be applied here for the other shader stages.

Reply:
I merged a more general fix #4296, so you can use that pattern here instead, like: `pShaderOp->CS = pShaderOp->GetString("CS67");`.